hibernate-core/src/test/java/org/hibernate/orm/test/idgen/userdefined/SimpleGenerator.java

```diff
@@ -0,0 +1,23 @@
+package org.hibernate.orm.test.idgen.userdefined;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.IdentifierGenerator;
+
+/**
+ * @author Yanming Zhou
+ */
+public class SimpleGenerator implements IdentifierGenerator {
```

Comment:
Probably better to make this test work for a generic `BeforeExecutionGenerator`.

Reply:
Done.